tests/integration/dualstack/common/traffic.go

```diff
@@ -0,0 +1,214 @@
+//go:build integ
+// +build integ
+
+//  Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"istio.io/istio/pkg/test"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/cluster"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/echo/deployment"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/components/prometheus"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/yml"
+	promMetric "istio.io/istio/tests/integration/telemetry"
+)
+
+type TrafficTestCase struct {
+	name string
+	// destination app
+	destinationApp string
+	// config can optionally be templated using the params src, dst (each are []echo.Instance)
+	config string
+
+	// Single call. Cannot be used with children.
+	call func(t test.Failer, options echo.CallOptions) echo.CallResult
+	// opts specifies the echo call options. When using RunForApps, the Target will be set dynamically.
+	opts echo.CallOptions
+
+	// setting cases to skipped is better than not adding them - gives visibility to what needs to be fixed
+	skip bool
+}
+
+func (c TrafficTestCase) Run(ctx framework.TestContext, namespace string) {
```

Comment:
For the most part, I think `tests/integration/pilot/common/` can support this. If there is a reason we re-implement it here, can you add a comment explaining why?

Reply:
For the most part it's the same. We do want to verify proxies are not sending http requests at the same time which is why I have a similar function: https://github.com/istio/istio/pull/38892/files#diff-8d1061ab02227bf02916079cac53e632843d46dcd82ce42a24e7ef20eeec6f89R75